redis/src/commands/cluster_scan.rs

```diff
@@ -0,0 +1,639 @@
+
+    /// Update the next node to be scanned.
+    ///
+    /// This method updates the scan state by updating the scanned slots map, determining the next slot to be scanned,
+    /// retrieving the node for that slot, and updating the slots in the scan map.
+    ///
+    /// # Arguments
+    ///
+    /// * `connection` - The connection to the Redis cluster.
+    ///
+    /// # Returns
+    ///
+    /// The updated ScanState.
+    async fn update_scan_state_and_get_next_node<C: ClusterInScan + ?Sized>(
+        &mut self,
+        connection: &C,
+    ) -> ScanState {
+        let _ = connection.refresh_slots().await;
+        let mut scanned_slots_map = self.scanned_slots_map;
+        // After each node we check if slots are missing, if so we cant know which slots has been scanned and which are new so we will skip updating the scanned_slots_map
+        let has_slots_of_node_changed = self.check_if_node_slots_changed(connection).await;
+        let new_node_epoch = connection
+            .get_node_epoch(&self.node_in_scan)
+            .await
+            .unwrap_or(0);
+        // If the node epoch changed it mean that some slots in the node are new, so we cant know which slots been there from the begining and which are new, or out and in later.
+        // In this case we will skip updating the scanned_slots_map and will just update the node and the cursor
+        if new_node_epoch != self.node_epoch || has_slots_of_node_changed {
+            return self
+                .update_scan_state_without_updating_scanned_map(connection)
+                .await;
+        }
+        // Update the scanned slots map with the slots in scan map
+        for (i, _) in self.scanned_slots_map.iter().enumerate() {
+            scanned_slots_map[i] = self.scanned_slots_map[i] | self.slots_in_scan_map[i];
+        }
+        // Get the next node to scan and its param base on the next slot set to 0 in the scanned_slots_map
+        let next_slot = self.get_next_slot(&scanned_slots_map).unwrap();
+        let new_node = if next_slot == END_OF_SCAN {
+            "".to_string()
+        } else {
+            connection.get_node_by_slot(next_slot).await
+        };
+        let new_epoch = connection.get_node_epoch(&new_node).await.unwrap_or(0);
+        let new_cursor = 0;
+        let mut slots_in_scan_map: SlotsBitsArray = [0; BITS_ARRAY_SIZE];
+        let slots_of_node = connection.get_slots_of_node(&new_node).await;
+        for slot in slots_of_node {
+            let word_index = slot as usize / BITS_PER_U64;
+            let bit_offset = slot as usize % BITS_PER_U64;
+            slots_in_scan_map[word_index] |= 1 << bit_offset;
+        }
+        ScanState::create(
+            new_cursor,
+            scanned_slots_map,
+            slots_in_scan_map,
+            new_node,
+            new_epoch,
+        )
+    }
+}
+
+// Implement the `ClusterInScan` trait for async `ClusterConnInner`
+#[async_trait]
+impl<C> ClusterInScan for Arc<InnerCore<C>>
```

Comment:
```suggestion
impl<C> ClusterInScan for Core<C>
```

by making crate::cluster_async::Core a pub(crate)